driver-core/src/main/com/mongodb/connection/ProxySettings.java

```diff
@@ -0,0 +1,343 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.connection;
+
+
+import com.mongodb.ClientEncryptionSettings;
+import com.mongodb.ConnectionString;
+import com.mongodb.annotations.Immutable;
+import com.mongodb.lang.Nullable;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+
+import static com.mongodb.assertions.Assertions.isTrue;
+import static com.mongodb.assertions.Assertions.isTrueArgument;
+import static com.mongodb.assertions.Assertions.notNull;
+
+/**
+ * This setting is only applicable when communicating with a MongoDB server using the synchronous variant of {@code MongoClient}.
+ * <p>
+ * This setting is furthermore ignored if:
+ * <ul>
+ *     <li>the communication is via {@linkplain com.mongodb.UnixServerAddress Unix domain socket}.</li>
+ *     <li>a {@link StreamFactoryFactory} is {@linkplain com.mongodb.MongoClientSettings.Builder#streamFactoryFactory(StreamFactoryFactory)
+ *     configured}.</li>
+ * </ul>
+ *
+ * @see SocketSettings#getProxySettings()
+ * @see ClientEncryptionSettings#getKeyVaultMongoClientSettings()}
+ * @see ClientEncryptionSettings#getKeyVaultMongoClientSettings()}.
```

Comment:
- `ClientEncryptionSettings#getKeyVaultMongoClientSettings()` is doubled, and the format is not correct.
- If we link to `ClientEncryptionSettings.getKeyVaultMongoClientSettings`, then we should also link to `AutoEncryptionSettingsgetKeyVaultMongoClientSettings`.

```suggestion
 * @see ClientEncryptionSettings#getKeyVaultMongoClientSettings()
 * @see AutoEncryptionSettings#getKeyVaultMongoClientSettings()
```